content/5_Plat/Sqrt.mdx

```diff
@@ -29,116 +29,171 @@ approaches. Both run in $\mathcal{O}((N+Q)\sqrt N)$ time.
 <!-- partitioning into sqrt blocks (each block can have some sort of data structure ...)
  -->
 
-See the first example in CPH.
+We partition the array into $\sqrt N)$ blocks, where each block covers an
+interval of size $\sqrt{N}$. For each block, there exists a corresponding
+`update` and `query` operation.
+
+**Update Queries:** $\mathcal{O}(1)$
+
+To update an element at location $x$, we can find the corresponding block using
+the formula $\frac{x}{\texttt{blocksz}}$, and apply the corresponding difference
+between the element currently stored at $x$, and the element we want to change
+it to.
+
+**Sum Queries:** $\mathcal{O}(\sqrt{N})$
+
+To perform a sum query from $[0\ldots r]$, calculate
+
+$$
+\sum_{i = 0}^{R} \texttt{blocks}[i] + \sum_{\lfloor R \rfloor \cdot \texttt{blocksz}}^r \texttt{nums}[i]
+$$
+
+where $\texttt{blocks}[i]$ represents the total sum of the $i$-th block, and $R$
+represents $\frac{r}{\texttt{blocksz}}$. Since each of these blocks are of size
+$\sqrt{N}$, there can be at most $\sqrt{N}$ blocks.
```

Comment:
well in your solution the blocks are of size floor(sqrt(N)), so there can be more than sqrt(N) blocks.